File: C++/FizzBuzz_v1.cpp
/* C++ で FizzBuzz */
/* printf関数、cout.write()でも出力可能 */

/* ヘッダ読み込み */
#include <iostream>

/* 指定された名前の名前空間を使う */
using namespace std;

int main(){

    for (int i = 0; i <= 100; i++){
        if(i % 3 == 0 && i % 5 == 0){
            std::cout << i << " FizzBuzz." << std::endl;
        }else if(i % 3 == 0){
            std::cout << i << " Fizz." << std::endl;
        }else if(i % 5 == 0){
            std::cout << i << " Buzz." << std::endl;
        }else{
            std::cout << i << std::endl;
        }
    }

    return 0;
}

File: Haskell/FizzBuzz_v1.hs
import Control.Monad

-- HaskellでFizzBuzz
-- 関数の利用 f(x, y)という形でなく、f x y という形で書く。


-- 変数iの型はIntと宣言
-- i :: Int

main = fizzbuzz 100

fizzbuzz max = do
    forM_ [0..max] $ \i -> do
        if (i `mod` 3 == 0 && i `mod` 5 == 0)
        then
            putStrLn $ (show i) ++ " FizzBuzz."
        else if (i `mod` 3 == 0)
        then
            putStrLn $ (show i) ++ " Fizz."
        else if (i `mod` 5 == 0)
        then
            putStrLn $ (show i) ++ " Buzz."
        else
            putStrLn (show i)
